### PR TITLE
Start to add in codes for inclusion of offline singles

### PR DESCRIPTION
**Pull request by GarethDaviesGW**
Starting to add offline singles functionality:

This first commit will almost definitely cause codeclimate to complain due to repeated code between this and coinc_findtrigs and coinc_statmap. I'll sort this when I can

The initial IFAR calculation method is bad (count number louder in foreground), and will never be larger than the chunk size. So hopefully will get an improved version in  - this is just to get the code into a shared space

### CODE REVIEW

File: bin/hdfcoinc/pycbc_multiifo_sngls_findtrigs [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+parser.add_argument("--ranking-statistic", choices=stat.statistic_dict.keys(),
+                    default='newsnr',
+                    help="The single-detector ranking statistic to calculate")
+parser.add_argument("--statistic-keywords", nargs='*',
+                    default=[],
+                    help="Provide additional key-word arguments to be sent to "
+                         "the statistic class when it is initialized. Should "
+                         "be given in format --statistic-keywords "
+                         "KWARG1:VALUE1 KWARG2:VALUE2 KWARG3:VALUE3 ...")
+parser.add_argument("--loudest-keep-values", type=str, nargs='*',
```
Comment (ahnitz) [inline bin/hdfcoinc/pycbc_multiifo_sngls_findtrigs]: This is unused no?

Comment (GarethDaviesGW) [reply bin/hdfcoinc/pycbc_multiifo_sngls_findtrigs]: yes thats right - I thought I'd managed to remove all unused arguments must have missed that one

**Comment by ahnitz** [comment]
@GarethDaviesGW Can you look at some of the CC issues? Some of them look like they should be addressed.

**Comment by GarethDaviesGW** [comment]
> @GarethDaviesGW Can you look at some of the CC issues? Some of them look like they should be addressed.

CC is saying that it is fixed - though most of these probably should have been caught before as it was just moving the code! - hopefully these fixes are acceptable

**Comment by GarethDaviesGW** [comment]
I'm not sure what the status is here for codebeat - it's saying that it has found some issues but hasn't decided whether it has failed or passed

**Comment by GarethDaviesGW** [comment]
well it's let me merge anyway

**Pull request merged**